zerver/actions/realm_settings.py

```diff
@@ -486,3 +487,71 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    # if realm is not deactivated, but changed the domain
+    if not acting_user:
+        return
+
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.REALM_DEACTIVATION_EMAIL_SENT,
+        event_time=event_time,
+        extra_data=orjson.dumps(
+            {
+                RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(realm),
+            }
+        ).decode(),
+    )
+    context_self = {
+        "realm_uri": realm.uri,
+        "realm_name": realm.name,
+        "event_date": event_time.date(),
+        "acting_user_name": acting_user.full_name,
+        "acting_user_mail": True,
+    }
+    context_others = {
+        "realm_uri": realm.uri,
+        "realm_name": realm.name,
+        "event_date": event_time.date(),
+        "acting_user_name": acting_user.full_name,
+        "acting_user_mail": False,
+    }
+    language = realm.default_language
+    owners = realm.get_human_owner_users()
+
+    deactivating_owner = None
+    other_owners = []
+    for owner in owners:
+        if owner == acting_user:
+            deactivating_owner = owner
+        else:
+            other_owners.append(owner)
```

Comment:
this logic can be simplified to;
```
deactivating_owner = acting_user
other_owners = owners.remove(deactivating_owner)
```
or using sets

Reply:
.remove() will not work for queryset most probably. 
I think if we could use .exclude() function and pass the uuid of deactivating user.
Thoughts?

Reply:
Yes, i meant something of that kind, instead of iterating and if/else logic.